### ISSUE TITLE
System.ArgumentException: 'Syntax node is not within syntax tree'

### ISSUE THREAD

**Issue by Hosch250** [dotnet/roslyn#18730]
**Version Used**: VS 15.1 (latest NuGet packages)

**Steps to Reproduce**:

1. Create a solution with analyzers
2. Create a WPF solution
3. Find all references to a (private) field in all partial classes
4. Call `semanticModel.GetSymbolnfo(referenceNode)`

**Expected Behavior**:

Ideally, this would return the symbol info, even if the declaration was in a different file.

**Actual Behavior**:

> System.ArgumentException: 'Syntax node is not within syntax tree'

Here is a GitHub repo with an MVCE (the link points to the line the exception happens at): [jubilant-broccoli](https://github.com/Hosch250/jubilant-broccoli/blob/master/VSIXProject1/FieldCanBeReadonlyAnalyzer.cs#L61)

**Comment by CyrusNajmabadi** [comment]
> Ideally, this would return the symbol info, even if the declaration was in a different file.

This is not how the semantic model works.  A semantic model is *explicitly* and *intentionally* tied to a single syntax-tree.  The point is that the semantic model acts as a lifetime construct that one can use to analyze a tree *efficiently*, letting the data be cleaned up when they no longer need it.

If we allowed GetSymbolInfo to work on nodes outside of hte current semantic model's tree, it would be "pit of fail" situation, whereby people would naively do this, and we would end up doing a tremendous amount of work that would then be thrown away.  

--

TLDR: if you want to ask about the SymbolInfo for a given Node, get the semantic model for that node first.  it's not hard:

```var correctSemanticModel = semanticModel.Compilation.GetSemanticModel(referenceNode.SyntaxTree);```

**Comment by Hosch250** [comment]
Thanks for the update, @CyrusNajmabadi. I initially suspected this was the case, but jmarolf told me to file it as a bug.
